chainmanager/types/params.go

```diff
@@ -0,0 +1,148 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/maticnetwork/heimdall/params/subspace"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+)
+
+// Default parameter values
+const (
+	DefaultTxConfirmationTime time.Duration = 6 * 14 * time.Second
+)
+
+// Parameter keys
+var (
+	KeyTxConfirmationTime = []byte("TxConfirmationTime")
+	KeyChainParams        = []byte("ChainParams")
+)
+
+var _ subspace.ParamSet = &Params{}
+
+// ChainParams chain related params
+type ChainParams struct {
+	BorChainID            string                  `json:"bor_chain_id" yaml:"bor_chain_id"`
+	MaticTokenAddress     hmTypes.HeimdallAddress `json:"matic_token_address" yaml:"matic_token_address"`
+	StakingManagerAddress hmTypes.HeimdallAddress `json:"staking_manager_address" yaml:"staking_manager_address"`
+	RootChainAddress      hmTypes.HeimdallAddress `json:"root_chain_address" yaml:"root_chain_address"`
+	StakingInfoAddress    hmTypes.HeimdallAddress `json:"staking_info_address" yaml:"staking_info_address"`
+	StateSenderAddress    hmTypes.HeimdallAddress `json:"state_sender_address" yaml:"state_sender_address"`
+
+	// Bor Chain Contracts
+	StateReceiverAddress hmTypes.HeimdallAddress `json:"state_receiver_address" yaml:"state_receiver_address"`
+	ValidatorSetAddress  hmTypes.HeimdallAddress `json:"validator_set_address" yaml:"validator_set_address"`
+}
+
+func (cp ChainParams) String() string {
+	return fmt.Sprintf(`
+	BorChainID: 									%s
+  MaticTokenAddress:            %s
+	StakingManagerAddress:        %s
+	RootChainAddress:             %s
+  StakingInfoAddress:           %s
+	StateSenderAddress:           %s
+	StateReceiverAddress: 				%s
+	ValidatorSetAddress:					%s`,
+		cp.BorChainID, cp.MaticTokenAddress, cp.StakingManagerAddress, cp.RootChainAddress, cp.StakingInfoAddress, cp.StateSenderAddress, cp.StateReceiverAddress, cp.ValidatorSetAddress)
```

Comment:
Can we format this?

Reply:
formatter is already applied, it is looking ok in the editor

Reply:
Gotcha